docs/website/docs/developers/debugging/llvm.md

```diff
@@ -0,0 +1,136 @@
+---
+hide:
+  - tags
+tags:
+  - GPU
+  - ROCm
+icon: octicons/bug-16
+---
+
+# LLVM debugging playbook
+
+This page aims to collect notes on how to debug or reduce issues that
+appear to arise from within LLVM itself and how to generate useful LLVM
+bug reports.
+
+This guide contains platform-independent notes applicable to both
+CPU and GPU compilation. Additional GPU-specific notes (such as how
+to perform binary substitutions in an AMD GPU context) are contained
+within [the GPU debugging playbook](./gpu.md).
+
+## Generating LLVM IR
+
+Wthen bisecting, reducing, or debugging an issue that might manifest within
+LLVM, it can be helpful to use the
+`--iree-hal-dump-executable-intermediates-to=[directory]` (or the more general
+`--iree-hal-dump-executable-files-to=[directory]`) flags to `iree-compile`
+or `iree-opt`. These flags will cause IREE to write out the compiled LLVM module
+to the specified directory so you can operate on it directly.
+
+Generally, there will be a `.linked` file, which contains the LLVM IR shortly after
+it was generated by MLIR (though after steps like bitcode library linking where
+applicable) and a `.optimized` file, which contains the IR after the `opt` passes
+have been run. The `.optimized` file may include reproduction instructions (if
+it doesn't, the relevant compiler plugin hasn't been uptated to add them).
+
+Similarly, the final generated assembly (a `.s` or `.rocmasm` or so on) may
+include reproduction instructions. Where those are present, they should be helpful
+in manually recreating the LLVM compilation so that you no longer have to
+route any changes through IREE.
```

Comment:
This duplicates some information at https://iree.dev/developers/general/developer-tips/#dumping-executable-files. Some repetition is okay, but you may want to cross-link "for more information about this, see ...").

I see that section is also still missing a content tab entry for `GPU - ROCm`